### DISCUSSION TITLE
test/regression  2 error

### DISCUSSION THREAD

**Discussion by tanchanlik** [The-OpenROAD-Project/OpenROAD-flow-scripts#551] [Q&A]
Hi all

I  compile test/regression  with below  erros  ( file attched)

check_charBuf *FAIL*
find_clock *FAIL*
simple_test *FAIL*
balance_levels *FAIL*
max_cap *FAIL*
aes *FAIL*

See /home/tancha/Desktop/OpenROAD/test/results for log files
------------------------------------------------------
Found 2 tool regression failures.

Can anyone help ?what wring with the process?

Thanks
Chan Lik
[messages.txt](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/9276849/messages.txt)
[messages.txt](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/9276850/messages.txt)

**Comment by maliberty** [comment]
Please say what commit and exactly what commands you are using that causes this problem.  Also the OS & compiler you are using would be helpful.

**Reply by tanchanlik** [reply]
./test/regression ( after  test the  tool)

Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.4 LTS
Release:	20.04
Codename:	focal

**Reply by tanchanlik** [reply]
I tried to  redo  ( del Openroad) still the same
attached log file
[regression.txt](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/9276868/regression.txt)

**Comment by maliberty** [comment]
The commit it in git?
How are you compiling the tool?
The nightly CI is passing so I'm trying to narrow down what is different.

**Reply by tanchanlik** [reply]
sorry  I am  new user
can explain what is commit it in git?

I start from 
git clone --recursive https://github.com/The-OpenROAD-Project/OpenROAD.git
refer to below  LINK  ( STEp by step)
https://www.cnblogs.com/ic-design/p/15891699.html

**Reply by maliberty** [reply]
git rev-parse HEAD

**Reply by tanchanlik** [reply]
tancha@tancha-HP-Desktop-M01-F0xxx:~/Desktop/OpenROAD$ git rev-parse HEAD
bde179a070c3993f22e3d130ed6aa8f6ade0a225

**Comment by maliberty** [comment]
I tried it under a docker ubuntu 20 image and I got no error.  Do you have docker available to try?

**Reply by tanchanlik** [reply]
I  did  install docker
but I need some help  how to   try?
what command?
Thanks

**Comment by maliberty** [comment]
docker pull openroad/ubuntu20-dev
./etc/DockerHelper.sh create -os=ubuntu20 -target=builder -compiler=gcc
./etc/DockerHelper.sh test -os=ubuntu20 -target=builder -compiler=gcc

[Taken from the top level Jenkinsfile]

**Reply by tanchanlik** [reply]
It  works PERFECT with above  test
Passed all tool regressions.
what should I do next to install in my system?

THANKS

**Reply by tanchanlik** [reply]
attached the datalog
[test.txt](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/9277030/test.txt)


THANKS

Chan Lik

**Comment by tanchanlik** [comment]
if i use

./etc/Build.sh 

./test/regression flow

same error  happen again
with docker is  fine

Thanks

Chan Lik

**Comment by maliberty** [comment]
You must have something wrong with you dependencies locally.  You can either run inside the container or fix the dependencies
